File: docs/_code/kafka_producer.cpp
#include <mofka/KafkaDriver.hpp>
#include <mofka/TopicHandle.hpp>
#include <iostream>

int main(int argc, char** argv) {

    if(argc != 3) {
        std::cerr << "Usage: "
                  << argv[0] << " <config-file> <topic>" << std::endl;
        return -1;
    }

    auto config_file = argv[1];
    auto topic_name = argv[2];

    try {

        mofka::KafkaDriver driver{config_file};

        mofka::TopicHandle topic = driver.openTopic(topic_name);

        mofka::Producer producer = topic.producer();
        for(size_t i = 0; i < 100; ++i) {
            producer.push(
                mofka::Metadata{R"({"x":42,"name":"bob"})"},
                mofka::Data{}
            );
        }
        producer.flush();

    } catch(const mofka::Exception& ex) {
        std::cerr << ex.what() << std::endl;
    }

    return 0;
}

File: docs/_code/kafka_producer.py
import sys
import pymargo.core
from pymargo.core import Engine
from mochi.mofka.client import KafkaDriver


def produce(config_file: str, topic_name: str):

    driver = KafkaDriver(config_file, engine)
    topic = driver.open_topic(topic_name)
    producer = topic.producer()

    for i in range(0, 100):
        producer.push(
            metadata={"x": i*42, "name": "bob"},
            data=bytes())

    producer.flush()


if __name__ == "__main__":

    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} <config-file> <topic>")
    config_file = sys.argv[1]
    topic_name = sys.argv[2]

    produce(config_file, topic_name)
